File: string/0482_license_key_formatting/0482_license_key_formatting.py
class Solution:
    def licenseKeyFormatting(self, S: str, K: int) -> str:
        
        S = S.upper().replace('-','')
        size = len(S)
        s1 = K if size%K==0 else size%K
        res = S[:s1]
        while s1<size:
            res += '-'+S[s1:s1+K]
            s1 += K
        return res


S = "5F3Z-2e-9-w"
K = 4
res = Solution().licenseKeyFormatting(S, K)
print(res)

File: string/0482_license_key_formatting/0482_license_key_formatting.go
package main

import (
	"fmt"
	"strings"
)

func licenseKeyFormatting(S string, K int) string {
	str := strings.ReplaceAll(S, "-", "")
	for i := len(str) - K; i > 0; i -= K {
		str = str[:i] + "-" + str[i:]
	}
	return strings.ToUpper(str)
}

func main() {
	S := "5F3Z-2e-9-w"
	K := 4
	res := licenseKeyFormatting(S, K)
	fmt.Println(res)
}
